networkx/generators/classic.py

```diff
@@ -706,6 +707,59 @@ def star_graph(n, create_using=None):
     return G
 
 
+@nodes_or_number([0, 1])
+@nx._dispatch(graphs=None)
+def tadpole_graph(m, n, create_using=None):
+    """Returns the Tadpole Graph; `C_m` connected to `P_n`.
+
+    Parameters
+    ----------
+    m, n : int or iterable container of nodes (default = 0)
+        If an integer, nodes are from `range(m)` and `range(m,m+n)`.
+        If a container of nodes, those nodes appear in the graph.
+        Warning: m and n are not checked for duplicates and if present the
+        resulting graph may not be as desired. Make sure you have no duplicates.
+
+        The nodes for m appear in the cycle graph $C_m$ and the nodes
+        for n appear in the path $P_n$
+    create_using : NetworkX graph constructor, optional (default=nx.Graph)
+       Graph type to create. If graph instance, then cleared before populated.
+
+    Notes
+    -----
+    The 2 subgraphs are joined via an edge (m-1, m).
+    If n=0, this is merely a cycle graph.
+
+    """
+    m, m_nodes = m
+    M = len(m_nodes)
+    if M < 2:
+        raise NetworkXError("Invalid description: m should indicate at least 2 nodes")
+
+    n, n_nodes = n
+    if isinstance(m, numbers.Integral) and isinstance(n, numbers.Integral):
+        n_nodes = list(range(M, M + n))
+    N = len(n_nodes)
+
+    # the circle
+    G = cycle_graph(m_nodes, create_using)
+    if G.is_directed():
+        raise NetworkXError("Directed Graph not supported")
+
+    # the stick
+    G.add_nodes_from(n_nodes)
+    if N > 1:
+        G.add_edges_from(pairwise(n_nodes))
+
+    if len(G) != M + N:
+        raise NetworkXError("Nodes must be distinct in containers m and n")
+
+    # connect circle to stick
+    if M > 0 and N > 0:
+        G.add_edge(m_nodes[-1], n_nodes[0])
+    return G
+
+
```

Comment:
- why are directed graphs not supported?
- The doc_string says that distinct nodes are **not** checked.  That actually is a feature for those who wish to use it that way. So I suggest removing the check here.
- Using the `nx.add_path` function should work nicely too. And we can connect the path to the circle in a way that handles n=0.
```suggestion
    nx.add_path(G, [m_nodes[-1]] + n_nodes)


```

Reply:
> why are directed graphs not supported?
For consistency with other classic generators, like the lollipop graph and the star graph.

> The doc_string says that distinct nodes are not checked. That actually is a feature for those who wish to use it that way. So I suggest removing the check here.
Done

> Using the nx.add_path function should work nicely too. And we can connect the path to the circle in a way that handles n=0.
I have slightly modified your code, as `n_nodes` is sometimes a tuple instead of a list.